### PR TITLE
[Merged by Bors] - Initial stuff for Fluvio::connect_with_connector

### PR DESCRIPTION
**Pull request by simlay**
I also added in a wasm test. I've got mixed feelings about where it is now.

### CODE REVIEW

**Comment by sehz** [comment]
rustfmt is failing

**Review by sehz — APPROVED** [review]
LGTM

**Comment by sehz** [comment]
local cluster test is failing

**Comment by simlay** [comment]
> local cluster test is failing

"Expected browser binary location, but unable to find binary in default location, no 'moz:firefoxOptions.binary' capability provided, and no binary flag set on the command line"

@nacardin is firefox or chrome installed on the custom runner?

**Comment by sehz** [comment]
Suggest separating integration test as 2nd PR so code changes can go thru

File: src/client/src/fluvio.rs [hunk 1]
```diff
@@ -65,6 +65,12 @@ impl Fluvio {
     /// ```
     pub async fn connect_with_config(config: &FluvioConfig) -> Result<Self, FluvioError> {
         let connector = DomainConnector::try_from(config.tls.clone())?;
+        Self::connect_with_connector(connector, config).await
+    }
+    pub async fn connect_with_connector(
+        connector: DomainConnector,
+        config: &FluvioConfig,
+    ) -> Result<Self, FluvioError> {
```
Comment (nicholastmosher) [inline src/client/src/fluvio.rs:73]: Why are we adding this? And more importantly, what is the usage pattern here? Are we expecting typical users to leverage this or is it more of an implementation detail for the WASM library that users will not directly use?

**Review by sehz — COMMENTED** [review]
This allows client to use custom networking.  For example:  unix socket, websocket, QUIC, etc.

File: src/client/src/fluvio.rs [hunk 2]
```diff
@@ -204,3 +210,108 @@ fn check_platform_compatible(cluster_version: &Version) -> Result<(), FluvioErro
...
+        let topic = "wasm-test-produce-consume".to_string();
+        /*
+        let err = admin.create(topic, false, TopicSpec::default()).await;
+        tracing::error!("ERROR: {:?}", err);
+        assert!(err.is_ok())
+        */
+        let producer = client.topic_producer(topic.clone()).await;
+        assert!(producer.is_ok());
+        let producer = producer.unwrap();
+        let send = producer.send("foo", "bar").await;
```
Comment (simlay) [inline src/client/src/fluvio.rs:288]: This is a work around because #986 is still an issue.

**Comment by sehz** [comment]
will merge this after release

**Comment by sehz** [comment]
Move  changelog for this to 0.8.4

**Pull request closed without merging**